### DISCUSSION TITLE
Drag and drop onto "Add frames from sprite sheet"

### DISCUSSION THREAD

**Discussion by kenneth-leblanc** [godotengine/godot-proposals#7482] [Editor]
Dragging and dropping individual frames into AnimatedSprite2D is incredibly simple to use. However, if you have a sprite sheet you are forced to browse through the file folder structure to find the sprite sheet by clicking on "Add frames from sprite sheet".

My idea would be to allow a sprite sheet to be dropped onto the "Add frames from sprite sheet" and have it open that sprite sheet directly into the select frames dialog. I have an example gif attached of which button i am talking about.

Another method would be to have another button, or right click menu, that offers you to split a frame into multiple frames. That way you just drag and drop the sprite sheet into the animation sprite dialog, then split it into its separate frames.

Thoughts?

![Animation](https://github.com/godotengine/godot-proposals/assets/4635469/0be996bd-2471-42ce-b294-ce1bff04e0e4)

**Comment by Calinou** [comment]
Drag-and-dropping onto a button doesn't feel that great to me, as you need to aim over a small area. Instead, I think we should use a <kbd>Shift</kbd> modifier (with a text appearing in the area telling you to do so). This is what we do for drag-and-dropping scenes or textures from the FileSystem dock to the 2D editor.

**Reply by kenneth-leblanc** [reply]
So you are saying, when you drag and hover over the drop area, text would appear saying 
"Drag and drop to add as a single frame"
"Hold Shift when dropping to open in Select Frames Dialog" or "Hold Shift when dropping to split into multiple frames"

Or something like that?
I think that is a great solution!

**Reply by Calinou** [reply]
Yes, exactly :slightly_smiling_face:

**Reply by kenneth-leblanc** [reply]
Lets hope someone decides to implement this.

**Reply by KoBeWi** [reply]
A modifier already exists - Ctrl.

**Reply by kenneth-leblanc** [reply]
So it does, but clearly it isn't very discoverable. Can it be advertised like Calinou said?

**Reply by mathphye** [reply]
I will never figure out that by myself or in the documentation. 

Would be good to add something like Blender does, for better insights of shortcuts (also show the context-window first):

![image](https://github.com/godotengine/godot-proposals/assets/146298448/0a0e9aab-b1df-4216-a7d9-2ee301c66d12)

* I can't find a hint of this shortcut in docs: https://docs.godotengine.org/en/stable/tutorials/editor/default_key_mapping.html

**Reply by Calinou** [reply]
The easy way is to add a centered label in that box, like we do when you open the Animation bottom panel with no AnimationPlayer node selected.
